### PR TITLE
dev-libs/ocl-icd: Pass khronos-headers USE flag to econf

### PR DESCRIPTION
**Pull request by zavorka**
Package-Manager: Portage-2.3.66, Repoman-2.3.12

### CODE REVIEW

File: dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild [hunk 1]
```diff
@@ -26,7 +26,7 @@ src_prepare() {
 }
 
 multilib_src_configure() {
-	ECONF_SOURCE="${S}" econf --enable-pthread-once
+	ECONF_SOURCE="${S}" econf --enable-pthread-once $(use_enable khronos-headers official-khronos-headers)
```
Comment (titanofold) [inline dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild]: Split long lines.

Generally, you want to aim for no more than 80 characters on a line, but don't bend over backwards to do so.

Comment (zavorka) [reply dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild]: No problem. I refrained from splitting it for the sake of consistency as there still was a longer line below.

File: dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	replace-flags -Os -O2 # bug 646122
+
+	default
+	eautoreconf
+}
+
+multilib_src_configure() {
+	ECONF_SOURCE="${S}" \
+	econf \
+		--enable-pthread-once\
```
Comment (titanofold) [inline dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild]: Add a space before the slash. It makes it easier to see.

File: dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	OCL_DIR="/usr/$(get_libdir)/OpenCL/vendors/ocl-icd"
+	dodir ${OCL_DIR}/{,include}
+
+	# Install vendor library
+	mv -f "${D}/usr/$(get_libdir)"/libOpenCL* "${ED}${OCL_DIR}" || die "Can't install vendor library"
+
+	# Install vendor headers
+	if use khronos-headers; then
+		cp -r "${S}/khronos-headers/CL" "${ED}${OCL_DIR}/include" || die "Can't install vendor headers"
```
Comment (titanofold) [inline dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild]: Missed this one, too: Split the line.

File: dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	default
+
+	# Drop .la files
+	find "${D}" -name '*.la' -delete || die
+
+	OCL_DIR="/usr/$(get_libdir)/OpenCL/vendors/ocl-icd"
+	dodir ${OCL_DIR}/{,include}
+
+	# Install vendor library
+	mv -f "${D}/usr/$(get_libdir)"/libOpenCL* "${ED}${OCL_DIR}" || die "Can't install vendor library"
```
Comment (titanofold) [inline dev-libs/ocl-icd/ocl-icd-2.2.12-r2.ebuild]: Missed this in my first review: Split the line.

**Comment by zavorka** [comment]
> Just a couple minor tweaks.
> 
> Be sure to squash your commits.
> 
> Also, because I don't have Ruby installed, please include a build log via pastebin or something, so that I can see this will emerge successfully.

[raw ANSI](https://termbin.com/jwgv)
[xterm.js canvas](http://void.myamazing.horse/ocl-icd-build-log/)

**Comment by titanofold** [comment]
I emailed Patrick about this, and once we get a yay/nay response we'll act on this PR.

**Comment by Arfrever** [comment]
It seems that alternative fix was made in commit 780818b4b055f8c41d253e71cd9641322776e7a4.

**Pull request closed without merging**